### DISCUSSION TITLE
Is using app.get instead of app.use for middleware to apply only on GET routes acceptable?

### DISCUSSION THREAD

**Discussion by ilshm** [honojs/hono#3095] [Q&A] [answered]
Hi everyone,

I'm working on a project using Hono and I need to apply middleware only to GET routes. Normally, I would use app.use to apply middleware globally, but this middleware should only affect GET requests. To achieve this, I used app.get instead of app.use for the middleware.

Here’s a simplified version of my code:

``` js
import amazon from './routes/amazon.js'
import apiKeys from './routes/apiKeys.js'

const app = new Hono()

app.get('/*', apiKeyAuth({
  enforceAuth: true
}))

app.route('/', amazon)
app.route('/', apiKeys)
```

In this code, apiKeyAuth middleware is applied only to GET routes. My question is:

Is this an acceptable and recommended approach to limit middleware to GET routes in Hono? Are there any potential issues or better practices I should be aware of?

Any advice or suggestions would be greatly appreciated. Thank you!

**Comment by Meess** [comment, marked as answer]
That sounds like the recommended way, also [described in the docs](https://hono.dev/docs/guides/middleware#definition-of-middleware), so I that makes perfect sense!
<img width="734" alt="image" src="https://github.com/honojs/hono/assets/8512566/d6f7ccb7-8ff2-4b5e-b723-3b3e84f40c3f">
